Hunk #1: Fix -fno-common build failure.
Hunk #2: Switch to gtk+2: remove call to deprecated function.

Index: wmclockmon-config/edit.c
--- wmclockmon-config/edit.c.orig
+++ wmclockmon-config/edit.c
@@ -16,6 +16,8 @@ static GtkWidget *wid_time;
 static GtkWidget *wid_date;
 static GtkWidget *wid_mesg;
 
+char *newalarm;
+
 void set_data(GtkWidget *widget, GtkWidget *data) {
     char *wstat = gtk_entry_get_text(GTK_ENTRY(GTK_COMBO(wid_status)->entry));
     char *wtime = gtk_entry_get_text(GTK_ENTRY(wid_time));
@@ -76,7 +78,6 @@ void edit_dialog(const char *title,
                        bouton, TRUE, TRUE, 0);
     GTK_WIDGET_SET_FLAGS(GTK_WIDGET(bouton), GTK_CAN_DEFAULT);
     gtk_widget_grab_default(GTK_WIDGET(bouton));
-    gtk_widget_draw_default(GTK_WIDGET(bouton));
     gtk_widget_show(bouton);
 
     bouton = gtk_button_new_with_label("Cancel");
